### PR TITLE
GH-44667: [Archery] Only suppress Docker progress logs if `--quiet` is enabled

### PR DESCRIPTION
**Pull request by pitrou**
### Rationale for this change

https://github.com/apache/arrow/pull/44669 went overboard by always suppressing Docker progress output. This may be desirable on CI, to avoid flooding the logs, but is annoying when running `archery docker` locally.

### What changes are included in this PR?

Only suppress Docker progress output if `--quiet` is passed to `archery docker`.

### Are these changes tested?

Yes.

### Are there any user-facing changes?

Yes, this restores the previous behavior unless `--quiet` is passed.

* GitHub Issue: #44667

### CODE REVIEW

File: dev/archery/archery/docker/core.py [hunk 1]
```diff
@@ -43,6 +44,10 @@ def flatten(node, parents=None):
         raise TypeError(node)
 
 
+def verbosity_args():
+    return ['--quiet'] if log_ctx.quiet else []
```
Comment (kou) [inline dev/archery/archery/docker/core.py:48]: Hmm. I don't want to use `archer --quiet ...` in many places. We will miss `--quiet` easily in a new CI configuration.

How about checking `GITHUB_ACTIONS=true` environment variable?
See also: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables

```suggestion
    return ['--quiet'] if os.environ.get('GITHUB_ACTIONS') == 'true' else []
```

Comment (pitrou) [reply dev/archery/archery/docker/core.py:48]: Perhaps a general `running_on_ci()` function?

Comment (kou) [reply dev/archery/archery/docker/core.py:48]: It's a good idea.

**Comment by pitrou** [comment]
I opened https://github.com/apache/arrow/pull/44865 as an alternative solution.

**Comment by kou** [comment]
I close this in favor of https://github.com/apache/arrow/pull/44865.

**Pull request closed without merging**